File: LearnOpencv/9/python/main.py
import cv2
import numpy as np
import matplotlib.pyplot as plt

# 读取图像
img=cv2.imread("C:/github/lib/picture/9.png",cv2.IMREAD_GRAYSCALE)

# 定义结构元素
kernel=np.ones((5,5),np.uint8)

# 开运算（先腐蚀后膨胀）
transition1=cv2.erode(img,kernel,iterations=1)
result1=cv2.dilate(transition1,kernel,iterations=1)

# 闭运算（先膨胀后腐蚀）
transition2=cv2.dilate(img,kernel,iterations=1)
result2=cv2.erode(transition2,kernel,iterations=1)

#显示图像
cv2.imshow("Img",img)
cv2.imshow("Result1",result1)
cv2.imshow("Result2",result2)

#等待按键关闭
cv2.waitKey(0)

File: LearnOpencv/9/c++/main.cpp
#include <iostream>
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/opencv.hpp>
using namespace std;
using namespace cv;

int main()
{
    //读取图像
    Mat img=imread("C:/github/lib/picture/9.png",IMREAD_GRAYSCALE);

    //生成结构元素图像
    Mat kernel=getStructuringElement(2,Size(8,8));

    //开运算
    Mat result1;
    morphologyEx(img,result1,MORPH_OPEN,kernel);

    //闭运算
    Mat result2;
    morphologyEx(img,result2,MORPH_CLOSE,kernel);

    //显示图像
    imshow("Img",img);
    imshow("Open",result1);
    imshow("Close",result2);

    //等待按键关闭
    waitKey(0);
    return 0;
}